modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/ValidationManager.java

```diff
@@ -112,13 +113,28 @@ protected ValidationResult validateNode(
                     "Cluster tags do not match. Cluster tag: %s, cluster tag stored in CMG: %s",
                     clusterTag, state.clusterTag()
             ));
+        } else if (!isColocationEnabledMatched(isColocationEnabled(node))) {
+            return ValidationResult.errorResult(String.format(
+                    "Colocation enabled mode does not match. Joining node colocation mode is: %s, cluster colocation mode is: %s",
+                    isColocationEnabled(node),
+                    isColocationEnabled(logicalTopology.getLogicalTopology().nodes().iterator().next())
+            ));
         } else {
             putValidatedNode(node);
 
             return ValidationResult.successfulResult();
         }
     }
 
+    private static boolean isColocationEnabled(LogicalNode node) {
```

Comment:
Can we unify these two methods into one and have something like: `boolean joiningNodeMatchesColocationMode` ?

Reply:
I don't think that it'll bring extra clarity, since isColocationEnabled is used in ValidationError generation.

Reply:
I didn't notice that, makes sense then